### PR TITLE
fix memory issues when running tests

### PR DESCRIPTION
**Pull request by mattcasey**
Also fixes 'too many open connections' error from Prisma :)

### CODE REVIEW

File: db.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
   || new PrismaClient({
-    //    log: ['query']
+  //  log: ['query']
   });
 
-// remember this instance of prisma in development to avoid too many clients
-if (process.env.NODE_ENV === 'development') {
-  global.prisma = prisma;
-}
+if (process.env.NODE_ENV !== 'production') global.prisma = prisma;
```
Comment (mattcasey) [inline db.ts:15]: this brings back the singleton for all modes except production and uses code provided by prisma

File: jest.config-server.ts [hunk 2]
```diff
@@ -9,158 +9,28 @@ import createJestConfig from 'testing/createJestConfig';
  * https://nextjs.org/docs/testing
  * */
 export const jestConfig = {
-  // All imported modules in your tests should be mocked automatically
```
Comment (mattcasey) [inline jest.config-server.ts:12]: removed a bunch of stuff we don't use to make it easier to see the config all at once. there's links to documentation which i tihnk is enough

File: testing/setupDatabase.ts [hunk 3]
```diff
@@ -228,9 +228,3 @@ export function createBlock (options: Partial<Block> & Pick<Block, 'createdBy' |
     }
   });
 }
-
-export default async function seedDatabase () {
```
Comment (mattcasey) [inline testing/setupDatabase.ts:232]: removed dead code

File: jest.config-server.ts (hunk 2 above)
Comment (motechFR) [reply jest.config-server.ts:12]: Makes sense, we can always refer to jest docs.

Will also avoid outdated options being present as we upgrade in future

File: next.config.js [hunk 4]
```diff
@@ -71,23 +90,8 @@ const withBundleAnalyzer = require('@next/bundle-analyzer')({
...
-  '@fullcalendar/daygrid',
-  '@fullcalendar/interaction',
-  '@fullcalendar/react',
-  'react-dnd'
-]);
+const withTM = require('next-transpile-modules')(esmModules);
 
 module.exports = withBundleAnalyzer(withTM(config));
+
+module.exports.esmModules = esmModules;
```
Comment (motechFR) [inline next.config.js:97]: For my own understanding, could you please explain a bit further what's happening in this file with regards to ESM as well as this export you added?

File: testing/createJestConfig.ts [hunk 5]
```diff
@@ -12,7 +13,13 @@ interface JestConfig {
 export default async function overriddenConfig (_config: any) {
   return async function defaultExport () {
     const config: JestConfig = await createJestConfig(_config)();
-    config.transformIgnorePatterns = ['/.next/'];
+    config.transformIgnorePatterns = [
+      '/.next/',
+      // ignore all node_modules except for bangle.dev
+      `/node_modules/(?!(${esmModules.join('|')}))(.*)`,
```
Comment (motechFR) [inline testing/createJestConfig.ts:19]: Really cool trick for avoiding repetition!

**Review by motechFR — APPROVED** [review]
Nice one @mattcasey you've brought our tests from 8mins plus to 1 min 30, awesome stuff

File: jest.config-server.ts [hunk 6]
```diff
@@ -172,52 +42,14 @@ export const jestConfig = {
...
-
   testTimeout: 30000,
 
-  // A map from regular expressions to paths to transformers
+  // transform esm modules
+  transformIgnorePatterns: ['node_modules/(?!(\\@bangle\\.dev)/)'],
   transform: {
-    // eslint-disable-next-line no-useless-escape
-    // [`node_modules/(${esModules})/.*\.(js|jsx|mjs|cjs|ts|tsx)$`]: 'babel-jest'
-    // 'node_modules/@bangle\.dev/.*\.(js|jsx|mjs|cjs|ts|tsx)$': 'babel-jest'
```
[the remark this answers is not in the document]
Comment (mattcasey) [reply jest.config-server.ts:201]: ya it is redundant, forgot to remove it thanks!

**Pull request merged**